metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteReferencesService.java

```diff
@@ -0,0 +1,177 @@
+package com.linkedin.metadata.entity;
+
+import com.linkedin.data.DataComplex;
+import com.linkedin.data.DataList;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.ArrayDataSchema;
+import com.linkedin.data.schema.DataSchema;
+import com.linkedin.data.schema.PathSpec;
+import com.linkedin.data.schema.RecordDataSchema;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.entity.Aspect;
+import java.util.List;
+import java.util.ListIterator;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * Utility class that encapsulates the logic on how to modify a {@link RecordTemplate} in place to remove a single value
+ * following a concrete set of rules.
+ *
+ * It does this by a recursive tree traversal method, based on an aspect path provided for a given aspect.
+ * This so that we don't have to scan the entire aspect for the value to be removed and then figure out how to apply
+ * logical rules based on upstream optionality definitions.
+ *
+ * For more information see {@link #getAspectWithReferenceRemoved(String, RecordTemplate, DataSchema, PathSpec)}
+ */
+@Slf4j
+public class DeleteReferencesService {
```

Comment:
This class contains only static utility methods.

Therefore, its a utility class, not a service. Let's rename it DeleteEntityUtils?